fe/fe-core/src/main/java/com/starrocks/leader/ReportHandler.java

```diff
@@ -974,10 +974,52 @@ private static boolean isBackendReplicaHealthy(TTabletInfo backendTabletInfo) {
         return true;
     }
 
+    public static boolean migratableTablet(OlapTable tbl, long partitionId, long indexId, long tabletId) {
+        if (tbl.getKeysType() != KeysType.PRIMARY_KEYS) {
+            return true;
+        }
+
+        final SystemInfoService currentSystemInfo = GlobalStateMgr.getCurrentSystemInfo();
+        List<Backend> backends = currentSystemInfo.getBackends();
+        long maxLastSuccessReportTabletsTime = -1L;
+
+        for (Backend be : backends) {
+            long lastSuccessReportTabletsTime = TimeUtils.timeStringToLong(be.getBackendStatus().lastSuccessReportTabletsTime);
+            maxLastSuccessReportTabletsTime = Math.max(maxLastSuccessReportTabletsTime, lastSuccessReportTabletsTime);
+        }
+
+        Partition partition = tbl.getPartition(partitionId);
+        if (partition == null || partition.getVisibleVersionTime() > maxLastSuccessReportTabletsTime) {
+            // partition is null or tablet report has not been updated, unmigratable
+            return false;
+        }
+        MaterializedIndex idx = partition.getIndex(indexId);
+        if (idx == null) {
+            // index is null, unmigratable
+            return false;
+        }
+
+        LocalTablet tablet = (LocalTablet) idx.getTablet(tabletId);
```

Comment:
`tbl.getPartition` `partition.getIndex` `idx.getTablet` should be called in db read lock

Reply:
done